server/homebrew.api.js

```diff
@@ -234,30 +237,31 @@ const updateBrew = async (req, res)=>{
 	if(req.account) {
 		brew.authors = _.uniq(_.concat(brew.authors, req.account.username));
 	}
-
-	// Fetch the brew from the database again (if it existed there to begin with), and assign the existing brew to it
-	brew = _.assign(await HomebrewModel.findOne({ _id: brew._id }), brew);
-
-	if(!brew.markModified) {
-		// If it wasn't in the database, create a new db brew
-		brew = new HomebrewModel(brew);
+	// we need the tag type change in both getBrew and here to handle the case where we don't have a stub on which to perform the modification
+	if(typeof brew.tags === 'string') {
+		brew.tags = [];
 	}
 
-	brew.markModified('authors');
```

Comment:
I didn't see a reason to keep markModified, it functions without it. If there is one please feel free to share!

Reply:
I think you are right. markModified is only needed if Mongoose can't naturally detect that a field was updated (dates or untyped arrays), but "authors" and "systems" are both typed to arrays of strings in the Mongoose Schema so that should be fine.....